.github/workflows/gpu.yml

```diff
@@ -0,0 +1,71 @@
+name: GPU
+
+# on: workflow_call
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  ubuntu-build:
+    name: Build - Ubuntu
+
+    strategy:
+      matrix:
+        os: ['ubuntu-22.04']
+        build_type: [Release]
+        compiler: [{c: gcc, cxx: g++}]
+        shared_library: ['ON', 'OFF']
+    runs-on: ${{matrix.os}}
+
+    steps:
+      - name: Install apt packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y cmake libhwloc-dev libnuma-dev libjemalloc-dev libtbb-dev
+
+      - name: Checkout Level Zero
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          repository: oneapi-src/level-zero
+          path: level_zero
+  
+      - name: Build and Install Level Zero
+        run: |
+            cd level_zero
+            mkdir build
+            cd build
+            cmake .. -DCMAKE_INSTALL_PREFIX=/usr
```

Comment:
don't install in `/usr` if on self-hosted runner 😉 pls install locally, somewhere in workspace